### ISSUE TITLE
Indent false positive for multiline Boolean expression that contains a comparison

### ISSUE THREAD

**Issue by eygraber** [pinterest/ktlint#1726]
This is how Intellij formats this code (indentation set to 2 spaces):

```
internal class Foo {
  fun foo(): Boolean {
    val bar =
      true &&
        ""
          .length > 1

    return bar
  }
}
```

With this `.editorconfig`:

```
[*.{kt,kts}]
indent_size = 2
indent_style = space
```

ktlint formats it as:

```
internal class Foo {
  fun foo(): Boolean {
    val bar =
      true &&
        ""
        .length > 1

    return bar
  }
}
```

This does not happen if the comparison is the first term in the expression:

```
internal class Foo {
  fun foo(): Boolean {
    val bar =
      ""
        .length > 1 &&
        true

    return bar
  }
}
```

but does happen if the comparison isn't the last term in the expression:

```
internal class Foo {
  fun foo(): Boolean {
    val bar =
      true &&
        ""
          .length > 1 &&
        false

    return bar
  }
}
```

This is happening on ktlint 0.47.1 and 0.48.0-SNAPSHOT

I'm running `ktlint --debug --experimental Foo.kt` (the issue template says the use `--log-level=debug` but that fails saying it is an unknown option). Here's the log:

```
ktlint --debug --experimental Foo.kt 
00:32:25.669 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "baseline" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "checkstyle" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "json" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "format" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "html" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "plain" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "sarif" id.
00:32:25.670 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Initializing "plain" reporter with {verbose=false, color=false, color_name=DARK_GRAY, format=false}
00:32:25.998 [pool-1-thread-1] DEBUG com.pinterest.ktlint.core.internal.RuleRunnerSorter - Rules will be executed in order below (unless disabled):
           - standard:annotation, 
           - standard:annotation-spacing, 
           - standard:argument-list-wrapping, 
           - standard:chain-wrapping, 
           - standard:colon-spacing, 
           - standard:comma-spacing, 
           - standard:comment-spacing, 
           - standard:curly-spacing, 
           - standard:dot-spacing, 
           - standard:double-colon-spacing, 
           - standard:enum-entry-name-case, 
           - standard:filename, 
           - standard:final-newline, 
           - standard:import-ordering, 
           - standard:keyword-spacing, 
           - standard:modifier-order, 
           - standard:multiline-if-else, 
           - standard:no-blank-line-before-rbrace, 
           - standard:no-blank-lines-in-chained-method-calls, 
           - standard:no-consecutive-blank-lines, 
           - standard:no-empty-class-body, 
           - standard:no-empty-first-line-in-method-block, 
           - standard:no-line-break-after-else, 
           - standard:no-line-break-before-assignment, 
           - standard:no-multi-spaces, 
           - standard:no-semi, 
           - standard:no-trailing-spaces, 
           - standard:no-unit-return, 
           - standard:no-unused-imports, 
           - standard:no-wildcard-imports, 
           - standard:op-spacing, 
           - standard:package-name, 
           - standard:parameter-list-wrapping, 
           - standard:paren-spacing, 
           - standard:range-spacing, 
           - standard:spacing-around-angle-brackets, 
           - standard:spacing-between-declarations-with-annotations, 
           - standard:spacing-between-declarations-with-comments, 
           - standard:string-template, 
           - standard:unary-op-spacing, 
           - standard:wrapping, 
           - experimental:comment-wrapping, 
           - experimental:discouraged-comment-location, 
           - experimental:fun-keyword-spacing, 
           - experimental:function-return-type-spacing, 
           - experimental:function-start-of-body-spacing, 
           - experimental:function-type-reference-spacing, 
           - experimental:kdoc-wrapping, 
           - experimental:modifier-list-spacing, 
           - experimental:nullable-type-spacing, 
           - experimental:parameter-list-spacing, 
           - experimental:spacing-between-function-name-and-opening-parenthesis, 
           - experimental:type-argument-list-spacing, 
           - experimental:type-parameter-list-spacing, 
           - experimental:unnecessary-parentheses-before-trailing-lambda, 
           - experimental:function-signature, 
           - standard:indent, 
           - experimental:block-comment-initial-star-alignment, 
           - standard:trailing-comma-on-call-site, 
           - standard:max-line-length, 
           - standard:trailing-comma-on-declaration-site
/home/eli/workspace/Foo.kt:6:1: Unexpected indentation (10) (should be 8) (indent)
00:32:26.013 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - 366ms / 1 file(s) / 1 error(s)
```

**Comment by paul-dingemans** [comment]
I will schedule this problem for next release (0.49).

> the issue template says the use `--log-level=debug` but that fails saying it is an unknown option

This is indeed only valid for ktlint `0.48.x` and newer.

**Comment by paul-dingemans** [comment]
The invalid formatting of code below:
```
internal class Foo {
  fun foo(): Boolean {
    val bar =
      true &&
        ""
        .length > 1

    return bar
  }
}
```
seems already to been fixed in `0.48.0`. Unfortunatley, I can not identify the change that fixed it. So please let me know if `0.48.x` does not resolve the problem for you.
